contrib/beind/Dockerfile

```diff
@@ -0,0 +1,92 @@
+FROM docker.io/library/alpine:3.8 as balena
+
+ARG BALENA_VERSION=v18.9.7
+
+RUN apk add --no-cache \
+		ca-certificates
+
+# set up nsswitch.conf for Go's "netgo" implementation (which Docker explicitly uses)
+# - https://github.com/docker/docker-ce/blob/v17.09.0-ce/components/engine/hack/make.sh#L149
+# - https://github.com/golang/go/blob/go1.9.1/src/net/conf.go#L194-L275
+# - docker run --rm debian:stretch grep '^hosts:' /etc/nsswitch.conf
+RUN [ ! -e /etc/nsswitch.conf ] && echo 'hosts: files dns' > /etc/nsswitch.conf
+
+RUN set -eux; \
+	\
+	apkArch="$(apk --print-arch)"; \
+	case "$apkArch" in \
+		x86_64) dlArch='x86_64' ;; \
+		armhf) dlArch='armv7hf' ;; \
+		aarch64) dlArch='aarch64' ;; \
+		*) echo >&2 "error: unsupported architecture ($apkArch)"; exit 1 ;;\
+	esac; \
+	\
+	if ! wget -O balena-engine.tgz "https://github.com/balena-os/balena-engine/releases/download/${BALENA_VERSION}/balena-engine-${BALENA_VERSION}-${dlArch}.tar.gz"; then \
+		echo >&2 "error: failed to download balenaEngine ${BALENA_VERSION} for ${dlArch}"; \
+		exit 1; \
+	fi; \
+	\
+	tar --extract \
+		--file balena-engine.tgz \
+		--strip-components 1 \
+		--directory /usr/local/bin/ \
+	; \
+	rm balena-engine.tgz; \
+	\
+	balena-engine-daemon --version; \
+	balena-engine --version
+
+COPY modprobe.sh /usr/local/bin/modprobe
+COPY balena-entrypoint.sh /usr/local/bin/
+
+ENTRYPOINT ["/usr/local/bin/balena-entrypoint.sh"]
+CMD []
+
+
+FROM balena as beind
+
+# https://github.com/docker/docker/blob/master/project/PACKAGERS.md#runtime-dependencies
+RUN set -eux; \
+	apk add --no-cache \
+		btrfs-progs \
+		e2fsprogs \
+		e2fsprogs-extra \
+		iptables \
+		tini \
+		xfsprogs \
+		xz \
+# pigz: https://github.com/moby/moby/pull/35697 (faster gzip implementation)
+		pigz \
+	; \
+# only install zfs if it's available for the current architecture
+# https://git.alpinelinux.org/cgit/aports/tree/main/zfs/APKBUILD?h=3.6-stable#n9 ("all !armhf !ppc64le" as of 2017-11-01)
+# "apk info XYZ" exits with a zero exit code but no output when the package exists but not for this arch
+	if zfs="$(apk info --no-cache --quiet zfs)" && [ -n "$zfs" ]; then \
+		apk add --no-cache zfs; \
```

Comment:
Why do we need zfs?

Reply:
cleaned up the remaining extra dependencies